runtime/commonMain/src/kotlinx/serialization/modules/SerialModuleBuilders.kt

```diff
@@ -148,15 +149,24 @@ public class SerializersModuleBuilder internal constructor() : SerialModuleColle
         }
     }
 
+    @JvmName("registerSerializer") // Don't mangle method name for prettier stack traces
     internal fun <T : Any> registerSerializer(
         forClass: KClass<T>,
         serializer: KSerializer<T>,
         allowOverwrite: Boolean = false
     ) {
-        if (!allowOverwrite && forClass in class2Serializer) throw SerializerAlreadyRegisteredException(forClass)
+        if (!allowOverwrite) {
+            val previous = class2Serializer[forClass]
+            if (previous != null && previous != serializer)
```

Comment:
Still, these serializers would be tested for referential equality (which will likely fail for generic serializers, since they're instantiated each time). I remember we had an idea for comparing serializers by their descriptors

Reply:
Filed #677 to resolve it separately. Because it's likely that we need changes in the plugin as well. Let it be the way it is right now

Reply:
I propose do not call `.equals()` on KSerializer directly, only on `SerialDescriptor`

Reply:
Well, it will complicate the life of format authors, that's why I've created a separate issue for that.
Maybe suggestion to use `this.descriptor == (other as? KSerializer<*>).descriptor` would be better